iface/get.go

```diff
@@ -232,16 +312,14 @@ func Qualify(pkg *types.Package, imps []Import, usedImps *[]Import) types.Qualif
 }
 
 func ValidateType(typ types.Type) bool {
-	return validateType(typ, &[]types.Type{})
+	return validateType(typ, map[types.Type]bool{})
 }
 
-func validateType(typ types.Type, visited *[]types.Type) bool {
-	for _, t := range *visited {
-		if t == typ {
-			return true
-		}
+func validateType(typ types.Type, visited map[types.Type]bool) bool {
+	if visited[typ] {
+		return true
 	}
-	*visited = append(*visited, typ)
```

Comment:
My linter pointed out this loop could be a `slices.Contains` call, which made me realize it could be a map lookup.

Reply:
Or a `collections.Set` 🤷

Reply:
That was my first thought, but `mock` is GPL 3.0, and `nicheinc/collections` is currently private. 🤫

Reply:
🤯